opengrok-indexer/src/main/java/org/opengrok/indexer/web/Util.java

```diff
@@ -1388,62 +1390,72 @@ public static String createSlider(int offset, int limit, long size, HttpServletR
             int startingResult = offset - limit * (offset / limit % 10 + 1);
             int myFirstPage = startingResult < 0 ? 1 : startingResult / limit + 1;
             int myLastPage = Math.min(lastPage, myFirstPage + 10 + (myFirstPage == 1 ? 0 : 1));
-
-            // function taking the page number and appending the desired content into the final buffer
-            IntFunction<Void> generatePageLink = page -> {
-                int myOffset = Math.max(0, (page - 1) * limit);
-                if (myOffset <= offset && offset < myOffset + limit) {
-                    // do not generate anchor for current page
-                    buf.append("<span class=\"sel\">").append(page).append(SPAN_END);
-                } else {
-                    buf.append("<a class=\"more\" href=\"?");
-                    // append request parameters
-                    if (request != null && request.getQueryString() != null) {
-                        String query = request.getQueryString();
-                        query = query.replaceFirst(RE_Q_E_A_A_COUNT_EQ_VAL, "");
-                        query = query.replaceFirst(RE_Q_E_A_A_START_EQ_VAL, "");
-                        query = query.replaceFirst(RE_A_ANCHOR_Q_E_A_A, "");
-                        if (!query.isEmpty()) {
-                            buf.append(query);
-                            buf.append(AMP);
-                        }
-                    }
-                    buf.append(QueryParameters.COUNT_PARAM_EQ).append(limit);
-                    if (myOffset != 0) {
-                        buf.append(AMP).append(QueryParameters.START_PARAM_EQ).
-                                append(myOffset);
-                    }
-                    buf.append("\">");
-                    // add << or >> if this link would lead to another section
-                    if (page == myFirstPage && page != 1) {
-                        buf.append("&lt;&lt");
-                    } else if (page == myLastPage && myOffset + limit < size) {
-                        buf.append("&gt;&gt;");
-                    } else {
-                        buf.append(page);
-                    }
-                    buf.append("</a>");
-                }
-                return null;
+            String queryString = Optional.ofNullable(request)
+                    .map(HttpServletRequest::getQueryString)
+                    .map(query -> query.replaceFirst(RE_Q_E_A_A_COUNT_EQ_VAL, ""))
+                    .map(query -> query.replaceFirst(RE_Q_E_A_A_START_EQ_VAL, ""))
+                    .map(query -> query.replaceFirst(RE_A_ANCHOR_Q_E_A_A, ""))
+                    .orElse("");
+            IntConsumer generatePageLink = pageNumber -> {
```

Comment:
It is a bit confusing that the variable is named as the method below.

Reply:
Changed